test/tools/ossfuzz/abiV2Proto.proto

```diff
@@ -0,0 +1,104 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+syntax = "proto2";
+
+// Flattened specification of array dimension
+// If is_static is false, and this array dimension is contained
+// inside another dimension e.g., x[][2] ([2] being the outer dimension)
+// then `length` for this dimension is the length of the first dynamically
+// sized array. The other (n-1) lengths are unspecified
+message ArrayDimensionInfo {
+    required uint32 length = 1;
+    required bool is_static = 2;
+}
+
+message ArrayType {
+    oneof base_type_oneof {
+        IntegerType inty = 1;
+        FixedByteType byty = 2;
+        AddressType adty = 3;
+        StructType stty = 4;
```

Comment:
Adding ArrayType here does not work?

Reply:
Actually, it might work. I hadn't thought of it that way. I will leave it as is for now, since changing the spec would also require change to the visitor implementation.